app/src/main/java/de/danoeh/antennapod/dialog/FeedRefreshIntervalDialog.java

```diff
@@ -0,0 +1,161 @@
+        LayoutInflater inflater = activity.getLayoutInflater();
+
+        View view = inflater.inflate(R.layout.update_refresh_time_dialoxg, null);
+        builder.setView(view);
+
+        final String[] intervalSpinnerValues = context.getResources().getStringArray(R.array.update_intervall_values);
+        final String[] entries = appendHourUnit(activity, intervalSpinnerValues);
+        long currInterval = UserPreferences.getUpdateInterval();
+        int checkedItem = -1;
+
+        if (currInterval > 0) {
+            String currIntervalStr = String.valueOf(TimeUnit.MILLISECONDS.toHours(currInterval));
+            checkedItem = ArrayUtils.indexOf(intervalSpinnerValues, currIntervalStr);
+        }
+
+        ArrayAdapter<String> spinnerArrayAdapter = new ArrayAdapter<String>(context,
+                android.R.layout.simple_spinner_item, entries);
+        spinnerArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+        int hourOfDay = 7;
+        int minute = 0;
+        int[] updateTime = UserPreferences.getUpdateTimeOfDay();
+        if (updateTime.length == 2) {
+            hourOfDay = updateTime[0];
+            minute = updateTime[1];
+        }
+
+        int finalCheckedItem = checkedItem;
+        int finalHourOfDay = hourOfDay;
+        int finalMinute = minute;
+
+        Spinner spinner = (Spinner) view.findViewById(R.id.spinner);
+        RadioButton radioButtonInterval = (RadioButton) view.findViewById(R.id.intervalRadioButton);
+        RadioButton radioButtonTime = (RadioButton) view.findViewById(R.id.timeRadioButton);
+        RadioButton radioButtonDisable = (RadioButton) view.findViewById(R.id.disableButton);
+        RadioGroup radioGroupOne = (RadioGroup) view.findViewById(R.id.genderRadioGroup);
+        TimePicker timePicker = (TimePicker) view.findViewById(R.id.timePicker);
+        radioGroupOne.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(RadioGroup radioGroup, int i) {
+                switch (i) {
+                    case R.id.intervalRadioButton:
+
+                        spinner.setAdapter(spinnerArrayAdapter);
+                        spinner.setSelection(finalCheckedItem);
+                        spinner.setVisibility(View.VISIBLE);
+                        timePicker.setVisibility(View.GONE);
+                        break;
+                    case R.id.timeRadioButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.VISIBLE);
+
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                            timePicker.setHour(finalHourOfDay);
+                        }
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                            timePicker.setMinute(finalMinute);
```

Comment:
Please do this in one single condition. It also does not need to be done every time the radio buttons are changed. Could you please move view setup outside this listener and only change the visibility here?